### PR TITLE
add flag memory-reservation memory-swappiness

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

File: cmd/nerdctl/run.go [hunk 1]
```diff
@@ -145,7 +145,9 @@ func setCreateFlags(cmd *cobra.Command) {
 	// #region cgroups, namespaces, and ulimits flags
 	cmd.Flags().Float64("cpus", 0.0, "Number of CPUs")
 	cmd.Flags().StringP("memory", "m", "", "Memory limit")
+	cmd.Flags().String("memory-reservation", "", "Memory soft limit")
 	cmd.Flags().String("memory-swap", "", "Swap limit equal to memory plus swap: '-1' to enable unlimited swap")
+	cmd.Flags().Int64("memory-swappiness", -1, "Tune container memory swappiness (0 to 100) (default -1)")
```
Comment (Zheaoli) [inline cmd/nerdctl/run.go:150]: Maybe we need to update the README.md at the same time

File: README.md [hunk 2]
```diff
@@ -396,7 +396,9 @@ Cgroup flags:
 - :whale: `--cpuset-cpus`: CPUs in which to allow execution (0-3, 0,1)
 - :whale: `--cpuset-mems`: Memory nodes (MEMs) in which to allow execution (0-3, 0,1). Only effective on NUMA systems
 - :whale: `--memory`: Memory limit
+- :whale: `--memory-reservation`: Memory soft limit
 - :whale: `--memory-swap`: Swap limit equal to memory plus swap: '-1' to enable unlimited swap
+- :whale: `---memory-swappiness`: Tune container memory swappiness (0 to 100) (default -1)
```
Comment (junnplus) [inline README.md:401]:
```suggestion
- :whale: `--memory-swappiness`: Tune container memory swappiness (0 to 100) (default -1)
```

Comment (ningmingxiao) [reply README.md]: ok，thank you

**Review by junnplus — APPROVED** [review]
LGTM on green

File: cmd/nerdctl/run_cgroup_linux.go [hunk 3]
```diff
@@ -54,6 +60,16 @@ func generateCgroupOpts(cmd *cobra.Command, id string) ([]oci.SpecOpts, error) {
 		return nil, err
 	}
 
+	memSwapiness64, err := cmd.Flags().GetInt64("memory-swappiness")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_cgroup_linux.go:63]: nit
```suggestion
	memSwappiness64, err := cmd.Flags().GetInt64("memory-swappiness")
```

**Comment by AkihiroSuda** [comment]
Needs rebase

**Review by AkihiroSuda — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
Thanks

**Pull request merged**